src/components/Card.js

```diff
@@ -15,45 +28,60 @@ export default class Card {
   }
 
   _setEventListeners() {
-    this._cardEl
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
-      });
+    const likeButton = this._cardEl.querySelector(".card__like-button");
+    const deleteButton = this._cardEl.querySelector(".card__delete-button");
 
-    this._cardEl
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
+    if (likeButton) {
+      likeButton.addEventListener("click", () => {
+        this._handleLikeIcon(this);
+      });
+    }
+  
+    if (deleteButton) {
+      deleteButton.addEventListener("click", () => {
+        this._handleDeleteCard(this);
       });
+    }
 
+    if (this._cardImageElement) {
       this._cardImageElement.addEventListener("click", () => {
         if (this._handleImageClick) {
           this._handleImageClick(this._name, this._link);
         }
       });
-    
+    }
   }
-
-  _handleDeleteCard() {
-    this._cardEl.remove();
-    this._cardEl = null;
+  
+  renderCardLike() {
+    if(this._isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
   }
 
-  _handleLikeIcon() {
-    this._cardEl
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+  setIsLiked(isLiked) {
+    this._isLiked = isLiked;
+    this.renderCardLike();
   }
 
   generateCard() {
     this._cardEl = this._getTemplate();
+    this._cardEl.setAttribute('data-id', this._id);
     this._cardEl.querySelector(".card__name").textContent = this._name;
     this._cardImageElement = this._cardEl.querySelector(".card__image");
+    this._likeButton = this._cardEl.querySelector(".card__like-button");
     this._cardImageElement.src = this._link;
     this._cardImageElement.alt = this._name;
-
+    
     this._setEventListeners();
+    this.renderCardLike();
     return this._cardEl;
   }
+  
+
+  toggleLikeIcon() {
+    this._cardEl.querySelector(".card__like-button").classList.toggle("card__like-button_active");
```

Comment:
You've already found `this._likeButton ` in line 73. Please, use it everywhere


`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.